DO NOT MERGE integration-cli/docker_cli_logs_test.go: Wait()

kolyshkin opened the pull request:
To avoid a zombie apocalyplse, use cmd.Wait() to properly finish
the processes we spawn with Start().

Found while investigating DockerSuite.TestLogsFollowSlowStdoutConsumer
failure on ARM (see https://github.com/moby/moby/pull/34550#issuecomment-324937936).

kolyshkin committed:
integration-cli/docker_cli_logs_test.go: Wait()

To avoid a zombie apocalyplse, use cmd.Wait() to properly finish
the processes we spawn with Start().

Found while investigating DockerSuite.TestLogsFollowSlowStdoutConsumer
failure on ARM (see
https://github.com/moby/moby/pull/34550#issuecomment-324937936).

[v2: don't expect no error from Wait() when process is killed]

Signed-off-by: Kir Kolyshkin <kolyshkin@gmail.com>

dnephin commented:
Looks like this causes the test to timeout. I think normally the `logs` process follows, so it doesn't exit.

Maybe this should be a `defer func( kill; wait; )` ?

kolyshkin committed:
Debug for TestLogsFollowSlowStdoutConsumer

Signed-off-by: Kir Kolyshkin <kolyshkin@gmail.com>

kolyshkin committed:
TestLogsSinceFutureFollow: tinker with

Signed-off-by: Kir Kolyshkin <kolyshkin@gmail.com>

vieux commented:
@dnephin PTAL

dnephin commented on daemon/graphdriver/overlay/copy.go:
```diff
@@ -157,9 +157,10 @@ func copyDir(srcDir, dstDir string, flags copyFlags) error {
-			aTime := time.Unix(stat.Atim.Sec, stat.Atim.Nsec)
-			mTime := time.Unix(stat.Mtim.Sec, stat.Mtim.Nsec)
+			aTime := time.Unix(int64(stat.Atim.Sec), int64(stat.Atim.Nsec))
+			mTime := time.Unix(int64(stat.Mtim.Sec), int64(stat.Mtim.Nsec))
```
:+1:

dnephin commented on hack/make/.integration-test-helpers:
```diff
@@ -25,7 +25,7 @@ run_test_integration() {
 
 	(
-		flags="-check.v -check.timeout=${TIMEOUT} -test.timeout=360m $TESTFLAGS"
+		flags="-check.v -check.timeout=${TIMEOUT} -test.timeout=360m -check.f TestLogsFollowSlow $TESTFLAGS"
```
Looks like this needs to be removed.

You can set this in `TESTFLAGS` instead of modidying this file:

`TESTFLAGS="-check.f TestLogsFollowSlow"`

dnephin commented on hack/test/unit:
```diff
@@ -10,6 +10,8 @@
 #    TESTDIRS="./pkg/term" hack/test/unit
 #
+echo 'Skipped ;)'
+exit 0
```
Remove!

vdemeester replied:
:joy:

dnephin reviewed (commented):
Looks like the tests aren't actually running in CI because of the changes to the scripts.

kolyshkin commented:
I didn't mean it for review, I'm just playing with some tests on ARM. Sorry for the confusion.

kolyshkin committed:
Fix fetching LVM2 sources

Version 2.02.173 has disappeared, let's revert back to latest stable
one.

https://github.com/moby/moby/issues/34843

Signed-off-by: Kir Kolyshkin <kolyshkin@gmail.com>

kolyshkin commented:
OK this was a CI playground to investigate ARM-32bit failure with logs. It is no longer needed, except for the first patch which I have added to https://github.com/moby/moby/pull/34550 series.

So long, and thanks for all the fish! :dolphin:
